### DISCUSSION TITLE
Add Swatten SiH-5/6/8/10kW-TH profile

### DISCUSSION THREAD

**Discussion by DevMartinator** [davidrapan/ha-solarman#166] [Ideas]
Hello, thanks a lot for your work.
Could you add this inverter definition?
[Swatten.zip](https://github.com/user-attachments/files/17061534/Swatten.zip)
It's a 3ph hybrid inverter: https://www.swatten.com/show-3-5.html
Thanks in advance!

**Comment by davidrapan** [comment]
Hi @DevMartinator, can I ask you where you have this profile from? I noticed there are missing sensors for example for "Total Load Consumption". Or do you maybe have modbus documentation for this device?

**Reply by davidrapan** [reply]
nvm, I did found some documents for this device and it looks like those registers are unknown, yet. Too bad.

**I already added the file with necessary compliance changes.**

**Comment by DevMartinator** [comment]
Hi, thanks a lot for the integration. 
This is the modbus protocol I did receive from the manufacturer.
[Swatten inverter Modbus table.xlsx](https://github.com/user-attachments/files/17093300/Swatten.inverter.Modbus.table.xlsx)

**Reply by davidrapan** [reply]
Thank you.

Could you maybe try to find registers for **AC cumulative output power** and **AC daily output power**? Cause they are not present in the table but it looks like official Solarman App does provide those values.

**Comment by DevMartinator** [comment]
Hi, it looks these 2 parameters were removed from Solarman App. They are no more available.
![Solarman app screenshot](https://github.com/user-attachments/assets/f08818e6-18e4-411f-b3eb-b07195cdeeac)

**Reply by davidrapan** [reply]
Ah I see.. thank you for your help!

**Reply by DevMartinator** [reply]
Hello, Can you please modify the Swatten definition file?
The "Battery type" parameter is unknown.
I have to modify the file like I had previously to make it work.
`  - group: Settings
    items: 
    - name: "Battery type"
      class: "setting"
      state_class: "setting"
      uom: ""
      scale: 1
      rule: 2
      registers: [26101]
      icon: 'mdi:battery'
`
Thanks in advance!
